### PR TITLE
Add screenshots path from takeScreenshot\takeElementScreenshot actions into report (closes #2216)

### LINKED ISSUE #2216: Add screenshots path from takeScreenshot\takeElementScreenshot actions into report
### Are you requesting a feature or reporting a bug?

Requesting a feature.

### What is the current behavior?

If `path` option is specified for one of takeScreenshot actions in test, then `testRunInfo` parameter of `reportTestDone` reporter plugin's method receives only base screenshots path (specified in `screenshots` method of test runner).

### What is the expected behavior?

Information about screenshots locations could be obtained from `testRunInfo`.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system:
* testcafe version:
* node.js version:

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

File: test/functional/fixtures/api/es-next/take-screenshot/test.js [hunk 1]
```diff
@@ -98,6 +110,36 @@ describe('[API] t.takeScreenshot()', function () {
...
+                    var expected = config.browsers.reduce(function (value, browser) {
+                        value.push(getScreenshotInfo(browser.alias + '1.png'));
+                        value.push(getScreenshotInfo(browser.alias + '2.png'));
+
+                        return value;
+                    }, []);
+
+                    var sortFn = (a, b) => a.screenshotPath < b.screenshotPath ? -1 : 1;
+
+                    expect(result.screenshots.sort(sortFn)).eql(expected.sort(sortFn));
```
Comment (kirovboris) [inline test/functional/fixtures/api/es-next/take-screenshot/test.js]: Probably http://www.chaijs.com/api/bdd/#method_members will be more suitable?

File: src/reporter/index.js [hunk 2]
```diff
@@ -39,6 +40,7 @@ export default class Reporter {
             durationMs:     new Date() - reportItem.startTime,
             unstable:       reportItem.unstable,
             screenshotPath: reportItem.screenshotPath,
+            screenshots:    reportItem.screenshots,
```
Comment (kirovboris) [inline src/reporter/index.js]: We need to discuss in private the report format, I'm not sure about the current format for now.

Comment (AndreyBelym) [reply src/reporter/index.js]: At least it must be a key-value properties, not a simple string.

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
please review

File: src/reporter/index.js [hunk 3]
```diff
@@ -92,8 +94,10 @@ export default class Reporter {
             reportItem.errs     = reportItem.errs.concat(testRun.errs);
 
             if (!reportItem.pendingRuns) {
-                if (task.screenshots.hasCapturedFor(testRun.test))
+                if (task.screenshots.hasCapturedFor(testRun.test)) {
                     reportItem.screenshotPath = task.screenshots.getPathFor(testRun.test);
+                    reportItem.screenshots    = task.screenshots.getScreenshotInfo(testRun.test);
```
Comment (helen-dikareva) [inline src/reporter/index.js]: Should it be in plural: `getScreenshotsInfo`?

File: src/runner/test-run-controller.js [hunk 4]
```diff
@@ -6,6 +6,18 @@ import TestRun from '../test-run';
 // Const
 const QUARANTINE_THRESHOLD = 3;
 
+class Quarantine {
+    constructor () {
+        this.passedTimes   = [];
```
Comment (helen-dikareva) [inline src/runner/test-run-controller.js]: Now we store not only passed/failed test count, but also their indexes. So. let's name these properties `passedAttempts/failedAttempts`

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
review please

File: test/server/reporter-test.js [hunk 5]
```diff
@@ -39,20 +39,38 @@ describe('Reporter', function () {
...
-            meta:               {
+            name:        'fixture1test1',
+            fixture:     fixtureMocks[0],
+            skip:        false,
+            screenshots: [{
+                screenshotPath:      'screenshot1.png',
+                thumbnailPath:       'thumbnail1.png',
+                userAgent:           'chrome',
+                onFail:              false,
+                quarantineAttemptID: 2
```
Comment (VasilyStrelyaev) [inline test/server/reporter-test.js]: `quarantineAttemptID` -> `quarantineAttempt` - isn't it just a count, not an ID?

File: test/server/reporter-test.js [hunk 6]
```diff
@@ -39,20 +39,38 @@ describe('Reporter', function () {
...
-            screenshotExpected: true,
-            meta:               {
+            name:        'fixture1test1',
+            fixture:     fixtureMocks[0],
+            skip:        false,
+            screenshots: [{
+                screenshotPath:      'screenshot1.png',
+                thumbnailPath:       'thumbnail1.png',
+                userAgent:           'chrome',
+                onFail:              false,
```
Comment (VasilyStrelyaev) [inline test/server/reporter-test.js]: `onFail` -> `takenOnFail`

File: src/runner/test-run-controller.js [hunk 7]
```diff
@@ -43,40 +55,41 @@ export default class TestRunController extends EventEmitter {
...
+        if (this.testRun.addQuarantineInfo)
+            this.testRun.addQuarantineInfo(this.quarantine);
+
         return this.testRun;
     }
 
     async _endQuarantine () {
-        if (this.quarantine.attemptNumber > 1)
-            this.testRun.unstable = this.quarantine.passedTimes > 0;
+        if (this.quarantine.attempts > 1)
```
Comment (helen-dikareva) [inline src/runner/test-run-controller.js]: `this.quarantine.attempts.length > 1` ? Do we have tests for unstable/stable tests?)

File: test/server/reporter-test.js [hunk 8]
```diff
@@ -98,7 +116,10 @@ describe('Reporter', function () {
             test:              testMocks[0],
             unstable:          true,
             browserConnection: browserConnectionMocks[0],
-            errs:              []
+            errs:              [],
+            quarantine:        {
+                attempts: [['1', '2'], []]
```
Comment (VasilyStrelyaev) [inline test/server/reporter-test.js]: Should I check this also? Will users face this? If so, what does it mean?

File: test/server/reporter-test.js [hunk 9]
```diff
@@ -289,11 +317,22 @@ describe('Reporter', function () {
...
-                        durationMs:     74000,
-                        unstable:       true,
-                        skipped:        false,
-                        screenshotPath: '/screenshots/1445437598847'
+                        errs:       [],
+                        durationMs: 74000,
+                        unstable:   true,
+                        skipped:    false,
+                        quarantine: {
+                            1: { isPassed: false },
```
Comment (VasilyStrelyaev) [inline test/server/reporter-test.js]: `isPassed` -> `passed`

File: test/server/reporter-test.js (hunk 5 above)
Comment (AlexKamaev) [reply test/server/reporter-test.js]: not a count, ID

File: test/server/reporter-test.js (hunk 8 above)
Comment (AlexKamaev) [reply test/server/reporter-test.js]: no, this is only for testing

File: src/runner/test-run-controller.js (hunk 7 above)
Comment (AlexKamaev) [reply src/runner/test-run-controller.js]: thanks, looks like we don't

File: test/server/reporter-test.js (hunk 8 above)
Comment (VasilyStrelyaev) [reply test/server/reporter-test.js]: okay then, let it be

File: test/server/reporter-test.js (hunk 5 above)
Comment (VasilyStrelyaev) [reply test/server/reporter-test.js]: okaaaay

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**